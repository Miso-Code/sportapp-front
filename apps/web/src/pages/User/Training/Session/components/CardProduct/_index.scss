.card-product {
	&-title {
		width: 10rem;
		white-space: nowrap;
		overflow: hidden;
		text-overflow: ellipsis;

		@media (min-width: 600px) {
			width: 19.5rem;
		}
	}

	&-description {
		width: 100%;
		overflow: hidden;
		max-height: 2.5rem;
		-webkit-box-orient: vertical;
		display: block;
		display: -webkit-box;
		text-overflow: ellipsis;
		-webkit-line-clamp: 2;
		line-height: normal;
	}
}
