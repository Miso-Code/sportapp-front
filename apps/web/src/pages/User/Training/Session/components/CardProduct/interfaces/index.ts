export interface Props {
	image: string
	title: string
	description: string
	onClick?: () => void
}
