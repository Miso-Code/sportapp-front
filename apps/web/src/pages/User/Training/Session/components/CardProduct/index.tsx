import { CardActionArea } from '@mui/material'
import Box from '@mui/material/Box'
import Card from '@mui/material/Card'
import CardContent from '@mui/material/CardContent'
import CardMedia from '@mui/material/CardMedia'
import Typography from '@mui/material/Typography'
import { Props } from './interfaces'
import './_index.scss'

export default function CardProduct({
	title,
	description,
	image,
	onClick
}: Props) {
	return (
		<Card
			sx={{
				width: 345,
				'@media (min-width: 600px)': { width: 500 }
			}}
			className='card-product'>
			<CardActionArea
				onClick={onClick}
				sx={{
					display: 'flex',
					justifyContent: 'unset',
					alignItems: 'unset',
					height: '8rem'
				}}>
				<CardMedia
					className='card-product-image'
					component='img'
					sx={{
						width: 151,
						objectFit: 'cover',
						objectPosition: 'center'
					}}
					image={image}
					alt='Live from space album cover'
				/>
				<Box sx={{ display: 'flex', flexDirection: 'column' }}>
					<CardContent sx={{ flex: '1 0 auto' }}>
						<Typography
							noWrap
							className='card-product-title'
							component='p'
							variant='h5'>
							{title}
						</Typography>
						<Typography
							className='card-product-description'
							variant='subtitle1'
							color='text.secondary'
							component='p'>
							{description}
						</Typography>
					</CardContent>
				</Box>
			</CardActionArea>
		</Card>
	)
}
