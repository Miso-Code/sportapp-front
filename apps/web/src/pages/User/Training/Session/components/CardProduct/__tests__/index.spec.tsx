import CardProduct from '..'
import { render, RenderResult } from '@testing-library/react'

describe('CardProduct', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(
			<CardProduct
				description='description'
				image='image'
				title='title'
			/>
		)
	})

	it('should render the component', () => {
		expect(wrapper.container).toMatchSnapshot()
	})
})
