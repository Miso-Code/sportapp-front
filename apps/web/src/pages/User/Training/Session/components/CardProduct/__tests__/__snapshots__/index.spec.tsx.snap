// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardProduct should render the component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card-product css-11w0ezu-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-58rx-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <img
        alt="Live from space album cover"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img card-product-image css-1n4vbju-MuiCardMedia-root"
        src="image"
      />
      <div
        class="MuiBox-root css-j7qwjs"
      >
        <div
          class="MuiCardContent-root css-1qaznj3-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap card-product-title css-1961ak2-MuiTypography-root"
          >
            title
          </p>
          <p
            class="MuiTypography-root MuiTypography-subtitle1 card-product-description css-ecpxie-MuiTypography-root"
          >
            description
          </p>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
